src/resultsdisassemblypage.cpp

```diff
@@ -171,6 +173,55 @@ ResultsDisassemblyPage::ResultsDisassemblyPage(CostContextMenu* costContextMenu,
         showDisassembly();
     });
 
+    ui->searchWidget->hide();
+
+    auto searchNext = [this] {
+        // point to next entry
+        int offset = m_currentSearchIndex.isValid() ? m_currentSearchIndex.row() : 0;
+        m_sourceCodeModel->find(ui->searchEdit->text(), Direction::Forward, offset);
+    };
+
+    auto searchPrev = [this] {
+        // point prev entry, need to subtract 2 since row() is index + 1
+        int offset = m_currentSearchIndex.isValid() ? m_currentSearchIndex.row() - 2 : 0;
+        m_sourceCodeModel->find(ui->searchEdit->text(), Direction::Backward, offset);
+    };
+
+    auto findAction = KStandardAction::find(
+        this,
+        [this] {
+            ui->searchWidget->show();
+            ui->searchEdit->setFocus();
+        },
+        this);
+    ui->sourceCodeView->addAction(findAction);
+
+    auto findNextAction = KStandardAction::findNext(this, searchNext, this);
+    ui->searchWidget->addAction(findNextAction);
+    auto findPrevAction = KStandardAction::findPrev(this, searchPrev, this);
+    ui->searchWidget->addAction(findPrevAction);
+
+    connect(ui->searchEdit, &QLineEdit::returnPressed, findNextAction, &QAction::trigger);
+    connect(ui->searchButton, &QPushButton::clicked, findNextAction, &QAction::trigger);
+    connect(ui->nextResult, &QPushButton::clicked, findNextAction, &QAction::trigger);
+    connect(ui->prevResult, &QPushButton::clicked, findPrevAction, &QAction::trigger);
+
+    KColorScheme colorScheme;
+    connect(m_sourceCodeModel, &SourceCodeModel::resultFound, this, [this, colorScheme](const QModelIndex& index) {
+        auto palette = ui->searchEdit->palette();
+        m_currentSearchIndex = index;
+        palette.setBrush(QPalette::Text, colorScheme.foreground());
+        ui->searchEdit->setPalette(palette);
+        ui->sourceCodeView->setCurrentIndex(index);
+    });
+
+    connect(m_sourceCodeModel, &SourceCodeModel::noResultFound, this, [this, colorScheme]() {
+        auto palette = ui->searchEdit->palette();
```

Comment:
DRY: can this share code with the super similar slot above? something like a setCurrentSearchIndex which you then call from here with an invalid index?

also: why is m_currentSearchIndex not set here?